### ISSUE TITLE
[a11y]: invalid tabindex for toolbar-content

### ISSUE THREAD

**Issue by Thejashwini123** [carbon-design-system/carbon#19892]
### Package

@carbon/web-components

### Browser

Chrome

### Operating System

MacOS

### Package version

2.33.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

Accessibility violation in carbon web-components table-toolbar-content
`The tabbable element does not have a valid widget role`

<img width="3410" height="1940" alt="Image" src="https://github.com/user-attachments/assets/2bfbe905-5c69-414d-9ae4-e32e4bbb8fb5" />

### Reproduction/example

carbon web-components - dataTable -> Toolbar -> Playground -> scan accessibility

### Steps to reproduce

carbon web-components - dataTable -> Toolbar -> Playground -> scan accessibility

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
